src/bun.zig

```diff
@@ -8,18 +8,18 @@ const bun = @This();
 
 pub const Environment = @import("./env.zig");
 
-pub const use_mimalloc = true;
+pub const use_mimalloc = !Environment.enable_asan;
```

Comment:
what if ASAN is disabled for zig but enabled for C++? then it will leak and we will never know.

let's make this a build.zig option that is false whenever ENABLE_ASAN is true.

Reply:
that is how asan is setup

but that's not how mimalloc is setup

when c++ asan is enabled, we want mimalloc to be disabled, even if you build the zig code without asan.